### ISSUE TITLE
IOException during liveness counterexample generation

### ISSUE THREAD

**Issue by AdrienChampion** [tlaplus/tlaplus#597]
Unfortunately, I cannot share the specification on which this problem occurs. The part of the output I can share is below.

I can tell you that some seed values do not trigger the exception, without touching the specification at all.

Let me know if I can do/run/modify anything to help track where the problem comes from.

-----

```text
TLC2 Version 2.16 of 31 December 2020 (rev: cdddf55)
Running breadth-first search Model-Checking with fp 2 and seed 42 with 6 workers on 6 cores with 3641MB heap and 64MB offheap memory (Mac OS X 10.16 x86_64, Oracle Corporation 1.8.0_281 x86_64, MSBDiskFPSet, DiskStateQueue).

[...]

Starting... (2021-03-25 17:03:03)
Implied-temporal checking--satisfiability problem has 1 branches.
Computing initial states...
Finished computing initial states: 1 distinct state generated at 2021-03-25 17:03:05.
Checking temporal properties for the current state space with 971 total distinct states at (2021-03-25 17:03:08)
Finished checking temporal properties in 00s at 2021-03-25 17:03:09
Progress(16) at 2021-03-25 17:03:09: 1,645 states generated (1,645 s/min), 971 distinct states found (971 ds/min), 237 states left on queue.
Progress(36) at 2021-03-25 17:03:56: 5,678 states generated, 2,907 distinct states found, 0 states left on queue.
Checking temporal properties for the complete state space with 2907 total distinct states at (2021-03-25 17:03:56)
Error: Temporal properties were violated.

Error: The following behavior constitutes a counter-example:

Error: TLC threw an unexpected exception.
This was probably caused by an error in the spec or model.
See the User Output or TLC Console for clues to what happened.
The exception was a java.io.IOException
: Read past end of file (increase with setLength)?
```

**Comment by lemmy** [comment]
If it is reproducible for some seeds, can you check if running with a single worker prevents the IOException?

**Comment by AdrienChampion** [comment]
> If it is reproducible for some seeds, can you check if running with a single worker prevents the IOException?

It does not :/

> Can you also check with older versions of TLC?

Sure, I will try that and get back to you.

-----

By the way, I found [this gitter thread](https://gitter.im/practical-tla-plus/community?at=5d9107cf086a72719e865494) where a potentially similar problem was reported. For convenience, here is the relevant part:

> tried dekker's algorithm with three threads (p107 version) and got this output:
>
> Temporal properties were violated.
> 
> TLC threw an unexpected exception.
> This was probably caused by an error in the spec or model.
> The error occurred when TLC was checking liveness.
> The exception was a java.io.IOException
> : Read past end of file (increase with setLength)?
>
> if I wanted to commit this stuff to a repo, would would I typically ignore as env-specific? I'd like to commit the specs and their models only, I think?
>
> here's a gist for the algorithm that produced that output: https://gist.github.com/autonomousapps/a8ebcb368fef211899fa6bda097c14f7

**Comment by lemmy** [comment]
Good catch!  Appears to be limited to macOS:

```shell
➜  Github597 uname -a
Darwin armbook 20.3.0 Darwin Kernel Version 20.3.0: Thu Jan 21 00:06:51 PST 2021; root:xnu-7195.81.3~1/RELEASE_ARM64_T8101 arm64
➜  Github597 java -version 
openjdk version "16" 2021-03-16
OpenJDK Runtime Environment Zulu16.28+11-CA (build 16+36)
OpenJDK 64-Bit Server VM Zulu16.28+11-CA (build 16+36, mixed mode)
➜  Github597 java -jar /Applications/TLA+\ Toolbox.app/Contents/Eclipse/tla2tools.jar dekker 
TLC2 Version 2.15 of Day Month 20?? (rev: c39cb21)
Warning: Please run the Java VM, which executes TLC with a throughput optimized garbage collector, by passing the "-XX:+UseParallelGC" property.
(Use the -nowarning option to disable this warning.)
Running breadth-first search Model-Checking with fp 128 and seed 7008211312854058528 with 1 worker on 8 cores with 2048MB heap and 64MB offheap memory [pid: 52340] (Mac OS X 11.2.3 aarch64, Azul Systems, Inc. 16 x86_64, MSBDiskFPSet, DiskStateQueue).
Parsing file /Users/markus/Documents/Github597/dekker.tla
Parsing file /private/var/folders/rn/dpqg4cgn7lg6k4_qwcx2jhvr0000gn/T/TLC.tla (jar:file:/Applications/TLA+%20Toolbox.app/Contents/Eclipse/tla2tools.jar!/tla2sany/StandardModules/TLC.tla)
Parsing file /private/var/folders/rn/dpqg4cgn7lg6k4_qwcx2jhvr0000gn/T/Integers.tla (jar:file:/Applications/TLA+%20Toolbox.app/Contents/Eclipse/tla2tools.jar!/tla2sany/StandardModules/Integers.tla)
Parsing file /private/var/folders/rn/dpqg4cgn7lg6k4_qwcx2jhvr0000gn/T/Naturals.tla (jar:file:/Applications/TLA+%20Toolbox.app/Contents/Eclipse/tla2tools.jar!/tla2sany/StandardModules/Naturals.tla)
Parsing file /private/var/folders/rn/dpqg4cgn7lg6k4_qwcx2jhvr0000gn/T/Sequences.tla (jar:file:/Applications/TLA+%20Toolbox.app/Contents/Eclipse/tla2tools.jar!/tla2sany/StandardModules/Sequences.tla)
Parsing file /private/var/folders/rn/dpqg4cgn7lg6k4_qwcx2jhvr0000gn/T/FiniteSets.tla (jar:file:/Applications/TLA+%20Toolbox.app/Contents/Eclipse/tla2tools.jar!/tla2sany/StandardModules/FiniteSets.tla)
Semantic processing of module Naturals
Semantic processing of module Sequences
Semantic processing of module FiniteSets
Semantic processing of module TLC
Semantic processing of module Integers
Semantic processing of module dekker
Starting... (2021-03-25 09:33:04)
Implied-temporal checking--satisfiability problem has 4 branches.
Computing initial states...
Computed 2 initial states...
Finished computing initial states: 3 distinct states generated at 2021-03-25 09:33:04.
Progress(25) at 2021-03-25 09:33:04: 4,356 states generated, 1,500 distinct states found, 0 states left on queue.
Checking 4 branches of temporal properties for the complete state space with 6000 total distinct states at (2021-03-25 09:33:04)
Error: Temporal properties were violated.

Error: The following behavior constitutes a counter-example:

Error: TLC threw an unexpected exception.
This was probably caused by an error in the spec or model.
See the User Output or TLC Console for clues to what happened.
The exception was a java.util.concurrent.ExecutionException
: java.io.IOException: Read past end of file (increase with setLength)?

```

```shell
markus@avocado:~/src/TLA/models/tlcbugs/Github597$ tlc dekker
TLC2 Version 2.15 of Day Month 20?? (rev: c39cb21)
Warning: Please run the Java VM, which executes TLC with a throughput optimized garbage collector, by passing the "-XX:+UseParallelGC" property.
(Use the -nowarning option to disable this warning.)
Running breadth-first search Model-Checking with fp 29 and seed 7936166605613410441 with 1 worker on 4 cores with 5952MB heap and 64MB offheap memory [pid: 201499] (Linux 5.4.0-70-generic amd64, Ubuntu 11.0.10 x86_64, MSBDiskFPSet, DiskStateQueue).
Parsing file /home/markus/src/TLA/_specs/models/tlcbugs/Github597/dekker.tla
Parsing file /tmp/TLC.tla (jar:file:/opt/toolbox/tla2tools.jar!/tla2sany/StandardModules/TLC.tla)
Parsing file /tmp/Integers.tla (jar:file:/opt/toolbox/tla2tools.jar!/tla2sany/StandardModules/Integers.tla)
Parsing file /tmp/Naturals.tla (jar:file:/opt/toolbox/tla2tools.jar!/tla2sany/StandardModules/Naturals.tla)
Parsing file /tmp/Sequences.tla (jar:file:/opt/toolbox/tla2tools.jar!/tla2sany/StandardModules/Sequences.tla)
Parsing file /tmp/FiniteSets.tla (jar:file:/opt/toolbox/tla2tools.jar!/tla2sany/StandardModules/FiniteSets.tla)
Semantic processing of module Naturals
Semantic processing of module Sequences
Semantic processing of module FiniteSets
Semantic processing of module TLC
Semantic processing of module Integers
Semantic processing of module dekker
Starting... (2021-03-25 09:34:52)
Implied-temporal checking--satisfiability problem has 4 branches.
Computing initial states...
Computed 2 initial states...
Finished computing initial states: 3 distinct states generated at 2021-03-25 09:34:52.
Progress(25) at 2021-03-25 09:34:53: 4,356 states generated, 1,500 distinct states found, 0 states left on queue.
Checking 4 branches of temporal properties for the complete state space with 6000 total distinct states at (2021-03-25 09:34:53)
Error: Temporal properties were violated.

Error: The following behavior constitutes a counter-example:

State 1: <Initial predicate>
/\ flag = (t1 :> FALSE @@ t2 :> FALSE @@ t3 :> FALSE)
/\ next_thread = t1
/\ pc = (t1 :> "P1" @@ t2 :> "P1" @@ t3 :> "P1")

State 2: <P1 line 60, col 13 to line 63, col 36 of module dekker>
/\ flag = (t1 :> FALSE @@ t2 :> TRUE @@ t3 :> FALSE)
/\ next_thread = t1
/\ pc = (t1 :> "P1" @@ t2 :> "P2" @@ t3 :> "P1")

State 3: <P1 line 60, col 13 to line 63, col 36 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> TRUE @@ t3 :> FALSE)
/\ next_thread = t1
/\ pc = (t1 :> "P2" @@ t2 :> "P2" @@ t3 :> "P1")

State 4: <P1 line 60, col 13 to line 63, col 36 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> TRUE @@ t3 :> TRUE)
/\ next_thread = t1
/\ pc = (t1 :> "P2" @@ t2 :> "P2" @@ t3 :> "P2")

State 5: <P2 line 65, col 13 to line 69, col 48 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> TRUE @@ t3 :> TRUE)
/\ next_thread = t1
/\ pc = (t1 :> "P2" @@ t2 :> "P2" @@ t3 :> "P2_1")

State 6: <P2 line 65, col 13 to line 69, col 48 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> TRUE @@ t3 :> TRUE)
/\ next_thread = t1
/\ pc = (t1 :> "P2_1" @@ t2 :> "P2" @@ t3 :> "P2_1")

State 7: <P2 line 65, col 13 to line 69, col 48 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> TRUE @@ t3 :> TRUE)
/\ next_thread = t1
/\ pc = (t1 :> "P2_1" @@ t2 :> "P2_1" @@ t3 :> "P2_1")

State 8: <P2_1 line 71, col 15 to line 75, col 50 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> TRUE @@ t3 :> TRUE)
/\ next_thread = t1
/\ pc = (t1 :> "P2" @@ t2 :> "P2_1" @@ t3 :> "P2_1")

State 9: <P2_1 line 71, col 15 to line 75, col 50 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> TRUE @@ t3 :> TRUE)
/\ next_thread = t1
/\ pc = (t1 :> "P2" @@ t2 :> "P2_1" @@ t3 :> "P2_1_1")

State 10: <P2_1 line 71, col 15 to line 75, col 50 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> TRUE @@ t3 :> TRUE)
/\ next_thread = t1
/\ pc = (t1 :> "P2" @@ t2 :> "P2_1_1" @@ t3 :> "P2_1_1")

State 11: <P2_1_1 line 77, col 17 to line 80, col 40 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> TRUE @@ t3 :> FALSE)
/\ next_thread = t1
/\ pc = (t1 :> "P2" @@ t2 :> "P2_1_1" @@ t3 :> "P2_1_2")

State 12: <P2_1_1 line 77, col 17 to line 80, col 40 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> FALSE @@ t3 :> FALSE)
/\ next_thread = t1
/\ pc = (t1 :> "P2" @@ t2 :> "P2_1_2" @@ t3 :> "P2_1_2")

State 13: <P2 line 65, col 13 to line 69, col 48 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> FALSE @@ t3 :> FALSE)
/\ next_thread = t1
/\ pc = (t1 :> "CS" @@ t2 :> "P2_1_2" @@ t3 :> "P2_1_2")

State 14: <CS line 92, col 13 to line 95, col 48 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> FALSE @@ t3 :> FALSE)
/\ next_thread = t1
/\ pc = (t1 :> "P3" @@ t2 :> "P2_1_2" @@ t3 :> "P2_1_2")

State 15: <P3 line 97, col 13 to line 101, col 27 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> FALSE @@ t3 :> FALSE)
/\ next_thread = t3
/\ pc = (t1 :> "P4" @@ t2 :> "P2_1_2" @@ t3 :> "P2_1_2")

State 16: <P2_1_2 line 82, col 17 to line 85, col 52 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> FALSE @@ t3 :> FALSE)
/\ next_thread = t3
/\ pc = (t1 :> "P4" @@ t2 :> "P2_1_2" @@ t3 :> "P2_1_3")

State 17: <P4 line 103, col 13 to line 105, col 48 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> FALSE @@ t3 :> FALSE)
/\ next_thread = t3
/\ pc = (t1 :> "P1" @@ t2 :> "P2_1_2" @@ t3 :> "P2_1_3")

State 18: <P1 line 60, col 13 to line 63, col 36 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> FALSE @@ t3 :> FALSE)
/\ next_thread = t3
/\ pc = (t1 :> "P2" @@ t2 :> "P2_1_2" @@ t3 :> "P2_1_3")

State 19: <P2 line 65, col 13 to line 69, col 48 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> FALSE @@ t3 :> FALSE)
/\ next_thread = t3
/\ pc = (t1 :> "CS" @@ t2 :> "P2_1_2" @@ t3 :> "P2_1_3")

State 20: <P2_1_3 line 87, col 17 to line 90, col 40 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> FALSE @@ t3 :> TRUE)
/\ next_thread = t3
/\ pc = (t1 :> "CS" @@ t2 :> "P2_1_2" @@ t3 :> "P2")

State 21: <CS line 92, col 13 to line 95, col 48 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> FALSE @@ t3 :> TRUE)
/\ next_thread = t3
/\ pc = (t1 :> "P3" @@ t2 :> "P2_1_2" @@ t3 :> "P2")

State 22: <P3 line 97, col 13 to line 101, col 27 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> FALSE @@ t3 :> TRUE)
/\ next_thread = t2
/\ pc = (t1 :> "P4" @@ t2 :> "P2_1_2" @@ t3 :> "P2")

State 23: <P4 line 103, col 13 to line 105, col 48 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> FALSE @@ t3 :> TRUE)
/\ next_thread = t2
/\ pc = (t1 :> "P1" @@ t2 :> "P2_1_2" @@ t3 :> "P2")

State 24: <P1 line 60, col 13 to line 63, col 36 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> FALSE @@ t3 :> TRUE)
/\ next_thread = t2
/\ pc = (t1 :> "P2" @@ t2 :> "P2_1_2" @@ t3 :> "P2")

State 25: <P2_1_2 line 82, col 17 to line 85, col 52 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> FALSE @@ t3 :> TRUE)
/\ next_thread = t2
/\ pc = (t1 :> "P2" @@ t2 :> "P2_1_3" @@ t3 :> "P2")

State 26: <P2 line 65, col 13 to line 69, col 48 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> FALSE @@ t3 :> TRUE)
/\ next_thread = t2
/\ pc = (t1 :> "P2_1" @@ t2 :> "P2_1_3" @@ t3 :> "P2")

State 27: <P2_1 line 71, col 15 to line 75, col 50 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> FALSE @@ t3 :> TRUE)
/\ next_thread = t2
/\ pc = (t1 :> "P2_1_1" @@ t2 :> "P2_1_3" @@ t3 :> "P2")

State 28: <P2_1_1 line 77, col 17 to line 80, col 40 of module dekker>
/\ flag = (t1 :> FALSE @@ t2 :> FALSE @@ t3 :> TRUE)
/\ next_thread = t2
/\ pc = (t1 :> "P2_1_2" @@ t2 :> "P2_1_3" @@ t3 :> "P2")

State 29: <P2 line 65, col 13 to line 69, col 48 of module dekker>
/\ flag = (t1 :> FALSE @@ t2 :> FALSE @@ t3 :> TRUE)
/\ next_thread = t2
/\ pc = (t1 :> "P2_1_2" @@ t2 :> "P2_1_3" @@ t3 :> "CS")

State 30: <CS line 92, col 13 to line 95, col 48 of module dekker>
/\ flag = (t1 :> FALSE @@ t2 :> FALSE @@ t3 :> TRUE)
/\ next_thread = t2
/\ pc = (t1 :> "P2_1_2" @@ t2 :> "P2_1_3" @@ t3 :> "P3")

State 31: <P2_1_3 line 87, col 17 to line 90, col 40 of module dekker>
/\ flag = (t1 :> FALSE @@ t2 :> TRUE @@ t3 :> TRUE)
/\ next_thread = t2
/\ pc = (t1 :> "P2_1_2" @@ t2 :> "P2" @@ t3 :> "P3")

State 32: <P3 line 97, col 13 to line 101, col 27 of module dekker>
/\ flag = (t1 :> FALSE @@ t2 :> TRUE @@ t3 :> TRUE)
/\ next_thread = t1
/\ pc = (t1 :> "P2_1_2" @@ t2 :> "P2" @@ t3 :> "P4")

State 33: <P2_1_2 line 82, col 17 to line 85, col 52 of module dekker>
/\ flag = (t1 :> FALSE @@ t2 :> TRUE @@ t3 :> TRUE)
/\ next_thread = t1
/\ pc = (t1 :> "P2_1_3" @@ t2 :> "P2" @@ t3 :> "P4")

State 34: <P2_1_3 line 87, col 17 to line 90, col 40 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> TRUE @@ t3 :> TRUE)
/\ next_thread = t1
/\ pc = (t1 :> "P2" @@ t2 :> "P2" @@ t3 :> "P4")

State 35: <P4 line 103, col 13 to line 105, col 48 of module dekker>
/\ flag = (t1 :> TRUE @@ t2 :> TRUE @@ t3 :> TRUE)
/\ next_thread = t1
/\ pc = (t1 :> "P2" @@ t2 :> "P2" @@ t3 :> "P1")

Back to state 4: <P1 line 60, col 13 to line 63, col 36 of module dekker>

Finished checking temporal properties in 00s at 2021-03-25 09:34:54
4356 states generated, 1500 distinct states found, 0 states left on queue.
The depth of the complete state graph search is 25.
Finished in 02s at (2021-03-25 09:34:54)
```

```tla
------------------------------- MODULE dekker -------------------------------
EXTENDS TLC, Integers
CONSTANT Threads

(*--algorithm dekker

variables 
  flag = [t \in Threads |-> FALSE],
  next_thread \in Threads;

define
  \* At most one thread is in the critical section at a time
  AtMostOneCritical == \A t1, t2 \in Threads:
    t1 /= t2 => ~(pc[t1] = "CS" /\ pc[t2] = "CS")
  
  \* All threads eventually reach the critical section
  Liveness == \A t \in Threads: <>(pc[t] = "CS")
end define;

fair process thread \in Threads
begin
  P1: flag[self] := TRUE;
  P2:
    while \E t \in Threads \ {self}: flag[t] do
      P2_1:
        if next_thread /= self then
          P2_1_1: flag[self] := FALSE;
          P2_1_2: await next_thread = self;
          P2_1_3: flag[self] := TRUE;
        end if;
    end while;
  CS: skip; \* CS means "critical section"
  P3: with t \in Threads \ {self} do
        next_thread := t;
      end with;
  P4: goto P1;
end process;

end algorithm;*)
\* BEGIN TRANSLATION
VARIABLES flag, next_thread, pc

(* define statement *)
AtMostOneCritical == \A t1, t2 \in Threads:
  t1 /= t2 => ~(pc[t1] = "CS" /\ pc[t2] = "CS")


Liveness == \A t \in Threads: <>(pc[t] = "CS")


vars == << flag, next_thread, pc >>

ProcSet == (Threads)

Init == (* Global variables *)
        /\ flag = [t \in Threads |-> FALSE]
        /\ next_thread \in Threads
        /\ pc = [self \in ProcSet |-> "P1"]

P1(self) == /\ pc[self] = "P1"
            /\ flag' = [flag EXCEPT ![self] = TRUE]
            /\ pc' = [pc EXCEPT ![self] = "P2"]
            /\ UNCHANGED next_thread

P2(self) == /\ pc[self] = "P2"
            /\ IF \E t \in Threads \ {self}: flag[t]
                  THEN /\ pc' = [pc EXCEPT ![self] = "P2_1"]
                  ELSE /\ pc' = [pc EXCEPT ![self] = "CS"]
            /\ UNCHANGED << flag, next_thread >>

P2_1(self) == /\ pc[self] = "P2_1"
              /\ IF next_thread /= self
                    THEN /\ pc' = [pc EXCEPT ![self] = "P2_1_1"]
                    ELSE /\ pc' = [pc EXCEPT ![self] = "P2"]
              /\ UNCHANGED << flag, next_thread >>

P2_1_1(self) == /\ pc[self] = "P2_1_1"
                /\ flag' = [flag EXCEPT ![self] = FALSE]
                /\ pc' = [pc EXCEPT ![self] = "P2_1_2"]
                /\ UNCHANGED next_thread

P2_1_2(self) == /\ pc[self] = "P2_1_2"
                /\ next_thread = self
                /\ pc' = [pc EXCEPT ![self] = "P2_1_3"]
                /\ UNCHANGED << flag, next_thread >>

P2_1_3(self) == /\ pc[self] = "P2_1_3"
                /\ flag' = [flag EXCEPT ![self] = TRUE]
                /\ pc' = [pc EXCEPT ![self] = "P2"]
                /\ UNCHANGED next_thread

CS(self) == /\ pc[self] = "CS"
            /\ TRUE
            /\ pc' = [pc EXCEPT ![self] = "P3"]
            /\ UNCHANGED << flag, next_thread >>

P3(self) == /\ pc[self] = "P3"
            /\ \E t \in Threads \ {self}:
                 next_thread' = t
            /\ pc' = [pc EXCEPT ![self] = "P4"]
            /\ flag' = flag

P4(self) == /\ pc[self] = "P4"
            /\ pc' = [pc EXCEPT ![self] = "P1"]
            /\ UNCHANGED << flag, next_thread >>

thread(self) == P1(self) \/ P2(self) \/ P2_1(self) \/ P2_1_1(self)
                   \/ P2_1_2(self) \/ P2_1_3(self) \/ CS(self) \/ P3(self)
                   \/ P4(self)

(* Allow infinite stuttering to prevent deadlock on termination. *)
Terminating == /\ \A self \in ProcSet: pc[self] = "Done"
               /\ UNCHANGED vars

Next == (\E self \in Threads: thread(self))
           \/ Terminating

Spec == /\ Init /\ [][Next]_vars
        /\ \A self \in Threads : WF_vars(thread(self))

Termination == <>(\A self \in ProcSet: pc[self] = "Done")

\* END TRANSLATION
=============================================================================
\* Modification History
\* Last modified Sun Sep 29 12:39:24 PDT 2019 by tony
\* Created Sun Sep 29 12:18:45 PDT 2019 by tony

---- CONFIG dekker -----
SPECIFICATION
    Spec
PROPERTIES
    Termination 
    Liveness
CONSTANT
    Threads = {t1,t2,t3}
=====
```

/cc @autonomousapps

**Comment by lemmy** [comment]
For now, it seems you can switch to Linux.

**Comment by AdrienChampion** [comment]
`<TODO: insert troll comment about macos and linux>`

Alright thanks!

**Comment by lemmy** [comment]
The exception only happens intermittently on Windows.  Please report back if you manage to trigger it on Linux.

**Comment by lemmy** [comment]
This bug also occurs on Linux!

**Comment by lemmy** [comment]
The last good version appears to be `TLC2 Version 2.07 of 1 June 2015` that is part of the TLA+ Toolbox 1.5.1 release.  I've made the 2.07 version of TLC and the 1.5.1 Toolbox release available at http://dl.tlapl.us/tlatoolbox/branches/1.5.1/.  Note that the old Toolbox might no longer launch on recent versions of macOS.

**Comment by lemmy** [comment]
Fix ba14c21 is included in the builds at https://nightly.tlapl.us.

**Comment by AdrienChampion** [comment]
I updated the IDE, ran TLC on my model, and got the actual counterexample with no `IOException`.

As far as I can tell, the problem seems to be fixed!

Thank you for your speed and efficiency 😸
